### PR TITLE
Upstream merge 0.3.1

### PR DESCRIPTION
**Pull request by waffletower-reify**
Merges  upstream changes against our cumulative develop branch

### CODE REVIEW

File: src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeEvent.java [hunk 1]
```diff
@@ -410,6 +413,12 @@ private List<Types.NestedField> icebergSchema(JsonNode eventSchema, String schem
             columnId += subSchema.size();
             break;
           default: //primitive types
+            if (fieldName.equals(partitionColumn)) {
+              // if it is the partition column, swap its type to timestamp
+              fieldType = "timestamptz";
+              // we also dont need to add a partition field, since it already exists.
+              addPartitionField = false;
+            }
```
Comment (dacreify) [inline src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeEvent.java:421]: Does this mean the feature only lets you partition on fields of this type? Seems a bit limited

File: src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeEvent.java [hunk 2]
```diff
@@ -419,9 +428,9 @@ private List<Types.NestedField> icebergSchema(JsonNode eventSchema, String schem
         }
       }
 
-      if (addSourceTsField) {
+      if (addPartitionField) {
         columnId++;
-        schemaColumns.add(Types.NestedField.optional(columnId, "__source_ts",
+        schemaColumns.add(Types.NestedField.optional(columnId, partitionColumn,
```
Comment (dacreify) [inline src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeEvent.java:433]: Hmm, so choosing a different partition column means no longer capturing `__source_ts`?

File: src/test/resources/json/debezium-annotated-schema.json [hunk 3]
```diff
@@ -56,6 +56,7 @@
     "__op": "r",
     "__table": "time",
     "__lsn": 33832960,
+    "__source_ms": 1596309869322,
     "__source_ts_ms": 1596309876678,
```
Comment (dacreify) [inline src/test/resources/json/debezium-annotated-schema.json:60]: Oh, maybe you still get it regardless of whether it's the partition column

**Review by dacreify — APPROVED** [review]

File: src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeEvent.java (hunk 2 above)
Comment (waffletower-reify) [reply src/main/java/com/getindata/kafka/connect/iceberg/sink/IcebergChangeEvent.java:433]: It is the default name.  The upstream change allows that column name to be changed.

**Pull request merged**